vj4/util/tools.py

```diff
@@ -18,5 +19,13 @@
         await module.ensure_indexes()
 
 
+@argmethod.wrap
+def extract_duplicate_key_errmsg(msg):
```

Comment:
Please don't do this, and I think mongodb doesn't guarantee that the error message format won't change.

Reply:
So we should give up a friendly error page, or check duplication with `find`.

Reply:
Do not give up :)

Reply:
But I really have no idea about that. I don't think there are other ways to do that.